internal/edge_cluster/edge_cluster_operations.go

```diff
@@ -0,0 +1,237 @@
+	result := make([]interface{}, 0)
+
+	for _, n := range newNodesRaw {
+		found := false
+		name := n.(map[string]interface{})["name"].(string)
+
+		for _, m := range currentNodes {
+			if name == *m.HostName {
+				found = true
+			}
+		}
+
+		if !found {
+			result = append(result, n)
+		}
+	}
+
+	return result
+}
+
+func getNodeSpec(node map[string]interface{}) *models.NsxTEdgeNodeSpec {
+	name := node["name"].(string)
+	clusterId := node["cluster_id"].(string)
+	tep1IP := node["tep1_ip"].(string)
+	tep2IP := node["tep2_ip"].(string)
+	tepGateway := node["tep_gateway"].(string)
+	tepVlan := int32(node["tep_vlan"].(int))
+
+	managementIP := node["management_ip"].(string)
+	managementGateway := node["management_gateway"].(string)
+
+	firstVdsUplink := node["first_nsx_vds_uplink"].(string)
+	secondVdsUplink := node["second_nsx_vds_uplink"].(string)
+
+	interRackCluster := node["inter_rack_cluster"].(bool)
+
+	nodeSpec := &models.NsxTEdgeNodeSpec{
+		ClusterID:          &clusterId,
+		EdgeNodeName:       &name,
+		EdgeTep1IP:         &tep1IP,
+		EdgeTep2IP:         &tep2IP,
+		EdgeTepGateway:     &tepGateway,
+		EdgeTepVlan:        &tepVlan,
+		ManagementGateway:  &managementGateway,
+		ManagementIP:       &managementIP,
+		FirstNsxVdsUplink:  firstVdsUplink,
+		SecondNsxVdsUplink: secondVdsUplink,
+		InterRackCluster:   &interRackCluster,
+		UplinkNetwork:      getUplinkNetworks(node),
+	}
+
+	return nodeSpec
+}
+
+func getUplinkNetworks(node map[string]interface{}) []*models.NsxTEdgeUplinkNetwork {
+	uplinks := node["uplink"].([]interface{})
+	specs := make([]*models.NsxTEdgeUplinkNetwork, 0, len(uplinks))
+
+	for _, u := range uplinks {
+		ip := u.(map[string]interface{})["interface_ip"].(string)
+		vlan := int32(u.(map[string]interface{})["vlan"].(int))
+		bgpPeersRaw := u.(map[string]interface{})["bgp_peer"].([]interface{})
+		spec := &models.NsxTEdgeUplinkNetwork{
+			UplinkInterfaceIP: &ip,
+			UplinkVlan:        &vlan,
+			BgpPeers:          getBgpPeers(bgpPeersRaw),
+		}
+
+		specs = append(specs, spec)
+	}
+	return specs
+}
+
+func getBgpPeers(bgpPeersRaw []interface{}) []*models.BgpPeerSpec {
+	peers := make([]*models.BgpPeerSpec, 0, len(bgpPeersRaw))
+
+	for _, p := range bgpPeersRaw {
+		ip := p.(map[string]interface{})["ip"].(string)
+		password := p.(map[string]interface{})["password"].(string)
+		asn := int64(p.(map[string]interface{})["asn"].(int))
+		peer := &models.BgpPeerSpec{
+			Asn:      &asn,
+			IP:       &ip,
+			Password: &password,
+		}
+
+		peers = append(peers, peer)
+	}
+
+	return peers
+}
+
+func getClusterProfileSpec(d *schema.ResourceData) *models.NsxTEdgeClusterProfileSpec {
+	var profileSpec *models.NsxTEdgeClusterProfileSpec = nil
+	profileType := d.Get("profile_type").(string)
+	if profileType == "CUSTOM" {
+		profileRaw := d.Get("profile").([]interface{})
+		if len(profileRaw) > 0 {
+			// there can be only one profile spec
```

Comment:
That wont be necessary, MaxItems is set on this property in the resource schema.
I'll double check just in case but I think it should throw an error if a user attempts to apply a config with more than 1 profile type